what4/src/What4/Expr/Builder.hs

```diff
@@ -5263,10 +5265,18 @@ instance IsExprBuilder (ExprBuilder t st fs) where
     | x == y = return $! truePred sym
     | otherwise = (floatIEEELogicBinOp FloatEq) sym x y
   floatNe sym x y = notPred sym =<< floatEq sym x y
-  floatFpEq = floatIEEELogicBinOp FloatFpEq
-  floatFpNe = floatIEEELogicBinOp FloatFpNe
-  floatLe = floatIEEELogicBinOp FloatLe
-  floatLt = floatIEEELogicBinOp FloatLt
+  floatFpEq sym x y
+    | x == y = notPred sym =<< floatIsNaN sym x
```

Comment:
I'm thinking that `floatFpEq` is not the dual of `floatFpNe` for NaN:

    let x = NaN
        y = NaN
    in assert $ (floatFpEq sym x y) /= (floatFpNe sym x y)

Should `floatFpEq` return a `truePred` if `x` and `y` are **both** `NaN`, or should `floatFpNe` return `truePred` for those inputs?

Reply:
OK, I didn't read ahead to the Interface changes... I see the issue now as described there.

Do we expect that `notPred . floatFpEq` is typically the more appropriate operation than `floatFpNe`?  If so, maybe we want to add another (ignored) parameter to `floatFpNe` to prevent people from using it without having to figure out why it isn't expressed naturally and therefore making it more likely they've seen the doc note?

Reply:
I think the documentation is probably sufficient.  I'd like to think that when presented with four variants of equality/non-equality tests, developers will read the docs to figure out which one to use.